core/miner/storage_fsm/impl/precommit_batcher_impl.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <boost/multiprecision/cpp_int.hpp>
+#include <chrono>
+#include <libp2p/protocol/common/asio/asio_scheduler.hpp>
+#include <libp2p/protocol/common/scheduler.hpp>
+#include <map>
+#include "api/full_node/node_api.hpp"
+#include "miner/storage_fsm/precommit_batcher.hpp"
+#include "miner/storage_fsm/types.hpp"
+#include "primitives/address/address.hpp"
+#include "primitives/types.hpp"
+#include "vm/actor/actor.hpp"
+
+namespace fc::mining {
+  using api::FullNodeApi;
+  using primitives::SectorNumber;
+  using libp2p::protocol::Scheduler;
+  using libp2p::protocol::scheduler::Handle;
+  using libp2p::protocol::scheduler::Ticks;
+  using primitives::address::Address;
+
+  class PreCommitBatcherImpl : public PreCommitBatcher {
+   public:
+    struct PreCommitEntry {
+      PreCommitEntry(const TokenAmount &number,
+                     const SectorPreCommitInfo &info);
+      PreCommitEntry() = default;
+      TokenAmount deposit;
+      SectorPreCommitInfo precommit_info;
+      PreCommitEntry &operator=(const PreCommitEntry &other) = default;
+    };
+
+    void setPreCommitCutoff(const ChainEpoch &current_epoch, const SectorInfo &sector_info);
+
+    static outcome::result<std::shared_ptr<PreCommitBatcherImpl>> makeBatcher(
+        const Ticks &max_wait,
+        std::shared_ptr<FullNodeApi> api,
+       const std::shared_ptr<Scheduler> & scheduler,
+        const Address &miner_address);
+
+    outcome::result<void> addPreCommit(const SectorInfo &sector_info,
+                                       const TokenAmount &deposit,
+                                       const SectorPreCommitInfo &precommit_info,
+                                       const PrecommitCallback &callback);
+
+    void forceSend() override;
+
+    outcome::result<CID> sendBatch();
+    PreCommitBatcherImpl(const Ticks &max_time,
+                         std::shared_ptr<FullNodeApi> api,
+                         const Address &miner_address,
+                         const Ticks &closest_cutoff);
```

Comment:
Not mandatory, but I suggest to declare the constructor at the beginning of the class

Reply:
From [google cpp style guide](https://google.github.io/styleguide/cppguide.html#Declaration_Order) ```prefer the following order: types (including typedef, using, enum, and nested structs and classes), constants, factory functions, constructors and assignment operators, destructor, all other methods, data members```

Reply:
done

Reply:
fixed